content/danielgormly/drone-plugin-kube/index.md

```diff
@@ -0,0 +1,55 @@
+---
+date: 2019-09-27T00:00:00+00:00
+title: Kubernetes
+author: danielgormly
+tags: [ kubernetes, deployment, configmap ]
+repo: danielgormly/drone-plugin-kube
+logo: kubernetes.svg
+image: danielgormly/drone-plugin-kube
+---
+
+Updates Kubernetes deployments from templates & configMaps from files. This plugin will either create or update existing resources dependent on their presence. It will wait for deployments before it progresses.
+
+Create or update deployment
+
+```yaml
+pipeline:
+- name: Deploy app
+  image: danielgormly/drone-plugin-kube:0.0.1
+  settings:
+    template: path/to/deployment.yaml # relative to repo root
+    ca: LS0tLS1... # BASE64 encoded string of the K8s CA cert
+    server: https://10.0.0.20:6443 # K8s master node address
+    token:
+      from_secret: kubernetes_token # Service account token to a service account that can manage deployments
+```
+
+Create or update config-map from a single file
+
+```diff
+pipeline:
+- name: Deploy app
+  image: danielgormly/drone-plugin-kube:0.0.1
+  settings:
+-    template: path/to/deployment.yaml
++    template: path/to/config-map.yaml
++    configmap_file: path/to/config-data.yaml
+    ca: LS0tLS1...
+    server: https://10.0.0.20:6443
+    token:
+      from_secret: kubernetes_token
+```
+
+# Secret Reference
+
+kubernetes_token
+: Kubernetes service account token (Not base64 encoded)
+
+# Parameter Reference
+
+ca
+: Base-64 encoded string of the K8s CA cert
+
+server
+: https://10.0.0.20:6443
+
```

Comment:
There must be more parameters... e.g. token, configmap_file?

Reply:
Good point thanks